File: backend/EmployeeBenefitsTracker.Api/Models/Employee.cs
namespace EmployeeBenefitsTracker.Api.Models;

public class Employee
{
    public int Id { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Department { get; set; } = string.Empty;
    public EmploymentStatus EmploymentStatus { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public List<BenefitEnrollment> BenefitEnrollments { get; set; } = new();
}

File: frontend/src/types/employee.ts
export type EmploymentStatus = 'Active' | 'Inactive';

export interface Employee {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  department: string;
  employmentStatus: EmploymentStatus;
  createdAt: string;
  updatedAt: string;
}

export interface PagedResult<T> {
  items: T[];
  page: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}

export interface CreateEmployeeRequest {
  firstName: string;
  lastName: string;
  email: string;
  department: string;
  employmentStatus: EmploymentStatus;
}

export type UpdateEmployeeRequest = CreateEmployeeRequest;

export interface EmployeeFilters {
  page?: number;
  pageSize?: number;
  search?: string;
  department?: string;
  benefitStatus?: string;
}
